src/main/java/anticope/rejects/utils/accounts/CustomYggdrasilLogin.java

```diff
@@ -71,50 +76,53 @@ public LocalYggdrasilMinecraftSessionService(YggdrasilAuthenticationService serv
             this.publicKey = getPublicKey(json.get("signaturePublickey").getAsString());
         }
 
-        private static PublicKey getPublicKey(String key) {
+        private static ServicesKeyInfo getPublicKey(String key) {
             key = key.replace("-----BEGIN PUBLIC KEY-----", "").replace("-----END PUBLIC KEY-----", "");
             try {
                 byte[] byteKey = Base64.getDecoder().decode(key.replace("\n", ""));
-                X509EncodedKeySpec spec = new X509EncodedKeySpec(byteKey);
-                KeyFactory factory = KeyFactory.getInstance("RSA");
-                return factory.generatePublic(spec);
-            } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+                return YggdrasilServicesKeyInfo.parse(byteKey);
+            } catch (IllegalArgumentException e) {
                 e.printStackTrace();
             }
             return null;
         }
 
-        @Override
```

Comment:
This has changed pretty significantly, but as far as I could tell, the old code was a copy-paste of the base class implementation with the `TextureUrlChecker.isAllowedTextureDomain()` check removed, and checking the signature against `publicKey` instead of `servicesKeySet` from the base class. I made the same changes when overriding the 1.20.4 method, but haven't tested it.